cli/command/image/build.go

```diff
@@ -250,16 +272,34 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		}
 	}
 
-	ctx := context.Background()
+	if options.stream && dockerfileCtx == nil {
+		f, err := os.Open(relDockerfile)
```

Comment:
minor nit: you can assign directly to `dockerfileCtx` here, no need to create a temporary variable.